### ISSUE TITLE
rustfmt will format comment even not covered by file-lines option

### ISSUE THREAD

**Issue by BusyJay** [rust-lang-nursery/rustfmt#1445]
Consider following file

```
struct A {
t: i64
}

#[cfg(test)]
mod tests {
    #[test]
    fn it_works() {
                                         // test
    }
}
```

If I format it using command ` rustfmt --write-mode=diff --file-lines '[{"file": "src/lib.rs", "range": [0, 2]}]' src/lib.rs`, only first two lines should be formatted, but I got these output:

```
Warning: Extra file listed in file_lines option '/tmp/testfmt/src/lib.rs'
Diff in src/lib.rs at line 1:
 struct A {⏎
-t: i64⏎
+    t: i64,⏎
 }⏎
 ⏎
 #[cfg(test)]⏎
Diff in src/lib.rs at line 6:
 mod tests {⏎
     #[test]⏎
     fn it_works() {⏎
-                                         // test⏎
+        // test⏎
     }⏎
 }⏎
```

It tries to format the comment at L9 too.

**Comment by kamalmarhubi** [comment]
As mentioned in https://github.com/rust-lang-nursery/rustfmt/issues/1514#issuecomment-302928915 I'm not so sure this is a bug as much as incomplete implementation. I had named the flag `--experimental-file-lines` but changed it to `--file-lines` during review.

**Comment by fintelia** [comment]
No, rustfmt shouldn't try to format code blatantly outside of file_lines. However, I believe this issue was resolved by https://github.com/rust-lang-nursery/rustfmt/pull/1510.

**Comment by topecongiro** [comment]
Closed via #1510.
